### PR TITLE
Ava UI: Various Fixes

### PR DESCRIPTION
**Pull request by AcK77**
This PR fixes :
- A screenshot path issue introduced in a recent PR (#4240), which currently doesn't allow users to take screenshots because of a malformed path.
- After the refactoring of `TitleUpdateManager` (#4276), if the JSON is malformed, it reset the internal field but not the file itself. If an update file doesn't exist anymore, it doesn't appear in the UI but still exist in the JSON file. This is now fixed.
- After the refactoring of `Renderer` (#4297), in `_wndProcDelegate` some fields are tested without taking care of the message send to the window, this is now fixed.

### CODE REVIEW

**Review by gdkchan — APPROVED** [review]

File: Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs [hunk 1]
```diff
@@ -223,4 +229,24 @@ public async void Add()
...
+
+            if (update == SelectedUpdate)
+            {
+                _titleUpdateWindowData.Selected = update.Path;
+            }
+        }
+
+        using FileStream titleUpdateJsonStream = File.Create(_titleUpdateJsonPath, 4096, FileOptions.WriteThrough);
+
+        titleUpdateJsonStream.Write(Encoding.UTF8.GetBytes(JsonHelper.Serialize(_titleUpdateWindowData, true)));
```
Comment (TSRBerry) [inline Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs:250]:
```suggestion
        File.WriteAllBytes(_titleUpdateJsonPath, Encoding.UTF8.GetBytes(JsonHelper.Serialize(_titleUpdateWindowData, true)));
```
Could this be shortened like this?

File: Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs [hunk 2]
```diff
@@ -102,6 +105,9 @@ private void LoadUpdates()
             AddUpdate(path);
         }
 
+        // NOTE: Save the list again to remove leftovers.
+        Save();
```
Comment (TSRBerry) [inline Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs:109]: It's probably not easier to remove the paths that don't exist in `TitleUpdates` from `_titleUpdateWindowData.Paths` instead, right?

**Review by TSRBerry — COMMENTED** [review]
Lgtm, nice to see these issues getting fixed. :D
Just some questions again, before I'll approve.

File: Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs (hunk 1 above)
Comment (AcK77) [reply Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs:250]: The using is here to use `WriteThrough` idk why it was added, maybe to be sure the real data will be written. I just kept the original code as-is.

File: Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs (hunk 2 above)
Comment (AcK77) [reply Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs:109]: That will not update the JSON file if a file doesn't exists and missing files path will stack in the JSON files.

File: Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs (hunk 1 above)
Comment (TSRBerry) [reply Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs:250]: Hmm, it looks fine to me to replace. `WriteThrough` just ensures cache is ignored and stuff is written to the disk directly. `WriteAllBytes` will open the file and close it after everything was written, so the changes should also be written to the disk at that point.

Comment (AcK77) [reply Ryujinx.Ava/UI/ViewModels/TitleUpdateViewModel.cs]: That should be fine yeah. Fixed!

**Review by TSRBerry — APPROVED** [review]

**Pull request merged**